tests/by-util/test_sort.rs

```diff
@@ -107,6 +107,32 @@ fn test_invalid_buffer_size() {
     }
 }
 
+#[test]
+fn test_legacy_plus_minus_accepts_when_modern_posix2() {
+    let (at, mut ucmd) = at_and_ucmd!();
+    at.write("input.txt", "aa\nbb\n");
+
+    ucmd.env("_POSIX2_VERSION", "200809")
+        .args(&["+1", "-1.5"])
+        .arg("input.txt")
+        .succeeds()
+        .stdout_is("aa\nbb\n");
+}
+
+#[test]
+fn test_legacy_plus_minus_rejected_in_traditional_range() {
```

Comment:
I'm unable to reproduce this test with GNU `sort` (v9.9) as I don't get an error:
```
$ _POSIX2_VERSION=200112 sort +1 -1.5 input.txt
aa
bb
```